components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt

```diff
@@ -140,9 +146,7 @@ internal class FilePicker(
      */
     @VisibleForTesting(otherwise = PRIVATE)
     internal fun onPermissionsGranted() {
-        store.consumePromptFrom(sessionId) { promptRequest ->
-            handleFileRequest(promptRequest as File, requestPermissions = false)
-        }
+        handleFileRequest(currentRequest as File, requestPermissions = false)
```

Comment:
Ah yes, that makes perfect sense. We were consuming the prompt request here and then it was no longer in the store once we hit `onActivityResult`. Would you mind adding a comment here to explain why we're consuming in `onPermissionDenied`, but not in `onPermissionGranted`? (Because then we will consume in `onActivityResult`).

Reply:
Should `currentRequest` be null able? As it will be there after we handle the request (`handleFileRequest`) should it be null after that?

Reply:
> Should `currentRequest` be nullable? At some point we should clean it up as it will persist there until a new request arrive

I used `currentRequest` as nullable and didn't check it for null just to be able to catch early any situations in which we would not cache the prompt as expected.. though there should be any.
To clean it up, just to reduce the memory footprint, I'll do it in `handleFileRequest`.

Reply:
nit: If we're setting it to `null` now, should we also set it to null if permissions were denied? We could do that in line 137 after either `onPermissionGranted` or `onPermissionDenied` have been called.

Reply:
Thanks!